src/VisualStudio/Core/Def/Implementation/Snippets/AbstractSnippetExpansionClient.cs

```diff
@@ -634,5 +1012,72 @@ internal bool TryGetSpanOnHigherBuffer(SnapshotSpan snapshotSpan, ITextBuffer ta
             span = default;
             return false;
         }
+
+        [NonCopyable]
+        private struct State
```

Comment:
being a mutable struct seems like it actually makes things more confusing TBH.  I can't imagine this needs to be a struct, or really benefits.  An immutable class assigned to a mutable field, or a mutable class assigned to a readonly field seems like it would be much clearer IMO.

Reply:
Mutable non-copyable struct and mutable class stored in a readonly field are functionally the same, but the former has less overhead. For now I have moved to the mutable class, but I'm pushing us towards a state where it's well understood that the latter is really just unnecessary overhead.

Reply:
➡️ The change here is covered by https://github.com/dotnet/roslyn/pull/49916#discussion_r550693829